### PR TITLE
Add new type definition for react-date-range.

### PR DESCRIPTION
**Pull request by junbong**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-date-range/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (mhegazy) [inline types/react-date-range/tsconfig.json]: this should be true for new packages.

File: types/react-date-range/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        "strictNullChecks": false,
```
Comment (mhegazy) [inline types/react-date-range/tsconfig.json]: this one too.

File: types/react-date-range/package.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "private": true,
+  "dependencies": {
+    "react-date-range": "^0.9.4"
```
Comment (mhegazy) [inline types/react-date-range/package.json]: First, I do not think you do not need this. Second, this package does not have its own definition files; keep in mind that these tests do not really run, these are just compilation checks to make sure that the test code "compiles" with no errors.

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by Junbong** [comment]
Okay. Thanks @mhegazy . I'll fix and push it again.

**Comment by Junbong** [comment]
@mhegazy Review again plz 🙂

**Pull request merged**